suite/src/__tests__/fast/model-correctness.test.ts

```diff
@@ -0,0 +1,58 @@
+import { describe, test, beforeAll, expect } from '@jest/globals'
+import { newCeramic } from '../../utils/ceramicHelpers.js'
+import { createDid } from '../../utils/didHelper.js'
+import { StreamID } from '@ceramicnetwork/streamid'
+import { Model } from '@ceramicnetwork/stream-model'
+import { ModelInstanceDocument } from '@ceramicnetwork/stream-model-instance'
+import { newModel, basicModelDocumentContent } from '../../models/modelConstants'
+import { CeramicClient } from '@ceramicnetwork/http-client'
+import { CommonTestUtils as TestUtils } from '@ceramicnetwork/common-test-utils'
+import { utilities } from '../../utils/common.js'
+
+const delay = utilities.delay
+const ComposeDbUrls = String(process.env.COMPOSEDB_URLS).split(',')
+// TODO : Update to the modelId created manually on all envs, if id cannot be the same then create an array for different envs
+const adminSeeds = String(process.env.COMPOSEDB_ADMIN_DID_SEEDS).split(',')
+const nodeSyncWaitTimeSec = 2
+
+describe('Model Integration Test', () => {
+  let ceramicNode1: CeramicClient
+  let ceramicNode2: CeramicClient
+  let modelId: StreamID
+  beforeAll(async () => {
+    const did1 = await createDid(adminSeeds[0])
+    let did2
+    if (adminSeeds[1]) {
+      did2 = await createDid(adminSeeds[1])
+    } else {
+      did2 = did1
+    }
+    ceramicNode1 = await newCeramic(ComposeDbUrls[0], did1)
+    ceramicNode2 = await newCeramic(ComposeDbUrls[1], did2)
+    let model = await Model.create(ceramicNode1, newModel)
+    TestUtils.waitForConditionOrTimeout(async () =>
```

Comment:
I believe this needs an await and should be for `ceramicNode2` since you are creating the model on `ceramicNode1`

Reply:
we are indexing the model on node 2. I added the await for node 1 so that the indexing call for node 2 and node 1 happens after we know for sure that the model has been created.

Reply:
Using `node1` we create the model. We can load it because during the `create` we store it on the associated `rust-ceramic node`. So we don't actually have to wait to see if we can load it. And indexing the model on `node1` should be able to happen right away. 

We also want to index the model on `node2`. But `node2` may not have the model yet because it's `rust-ceramic` node needs to sync with `node1's` `rust-ceramic` node via recon. Therefore we need to wait for `node2` to receive the model before indexing the model.

I'm not sure how its working now or how it worked without the `await`.

Reply:
I think both of the `TestUtils.waitForConditionOrTimeout` still needs an await as its an asynchronous function

Reply:
oops.. missed the await